### PR TITLE
Genesis file configuration

### PR DESCRIPTION
**Pull request by voddan**
Extracted setting up genesis file into configs; Added genesis files for indy-pool and STN

TODO: 
- [x] edit ./genesis/README.md
- [x] fix ReadmeExampleTest.kt

### CODE REVIEW

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/ReadmeExampleTest.kt [hunk 1]
```diff
@@ -100,34 +107,34 @@ class ReadmeExampleTest {
 
     @Test
     fun `grocery store example`() {
-        val ministry: StartedNode<*> = issuer
+        val ministry: StartedNode<InternalMockNetwork.MockNode> = issuer
         val alice: StartedNode<*> = alice
         val store: StartedNode<*> = bob
 
-// Each Corda node has a X500 name:
+        logger.info("Each Corda node has a X500 name:")
```
[the remark this answers is not in the document]
Comment (voddan) [reply cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/ReadmeExampleTest.kt]: I'll revert the loggers back to comments

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/ReadmeExampleTest.kt [hunk 2]
```diff
@@ -156,9 +163,20 @@ class ReadmeExampleTest {
...
+        val machine = startFlow(logic, newContext()).getOrThrow()
+
+        return object : FlowStateMachine<T> by machine {
+            override val resultFuture: CordaFuture<T>
+                get() {
+                    net.runNetwork()
+                    return machine.resultFuture
+                }
+        }
+    }
```
Comment (alkopnin) [inline cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/ReadmeExampleTest.kt]: Could we create separated file with the repeatable code (startFlow) from both tests?

File: indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/utils/PoolManager.kt [hunk 3]
```diff
@@ -1,24 +1,46 @@
...
+
+    fun openIndyPool(genesisFile: File, poolName: String = DEFAULT_POOL_NAME): Pool {
+        val pool = openIndyPools.getOrPut(poolName) {
+            val configName = createPoolLedgerConfigFromFile(genesisFile, poolName)
+
+            Pool.setProtocolVersion(2).get()
 
-    companion object {
-        private val mInstance: PoolManager = PoolManager()
+            val config = PoolJSONParameters.OpenPoolLedgerJSONParameter(null, null, null)
```
Comment (alkopnin) [inline indy-utils/src/main/kotlin/com/luxoft/blockchainlab/hyperledger/indy/utils/PoolManager.kt]: What are the params here? Could we use it somehow?

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+
+    /**
+     * List of all flows that may be initiated by a message
+     * */
+    val projectReciverFlows = listOf(
+            AssignPermissionsFlow.Authority::class,
+            CreatePairwiseFlow.Issuer::class,
+            IssueClaimFlow.Prover::class,
+            VerifyClaimFlow.Prover::class,
+            VerifyClaimFlow.Prover::class)
```
Comment (alkopnin) [inline cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: VerifyClaimFlow.Prover repeated twice

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt [hunk 5]
```diff
@@ -0,0 +1,145 @@
...
+     *
+     *     lateinit var issuer: StartedNode<MockNode>
+     *
+     *     @Before
+     *     createNodes() {
+     *         issuer = createPartyNode(CordaX500Name("Issuer", "London", "GB"))
+     *     }
+     * */
+    protected fun createPartyNode(legalName: CordaX500Name? = null): StartedNode<MockNode> {
+        val party = net.createPartyNode(legalName)
```
Comment (alkopnin) [inline cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: Why legalName is optional?

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt [hunk 6]
```diff
@@ -0,0 +1,145 @@
...
+                networkParameters = testNetworkParameters(maxTransactionSize = 10485760 * 5))
+
+        parties.clear()
+    }
+
+    @After
+    fun commonTearDown() {
+        try {
+            for (party in parties) {
+                for(service in projectServises) {
```
Comment (alkopnin) [inline cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: IndyService is just one service who uses IndyClient and has close function. 
If you will extend projectServices it breaks the logic. I suggest don't to use such construction and close the wallets in IndyService directly

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt (hunk 5 above)
Comment (voddan) [reply cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: It is optional in Corda, so who am I to change that?

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt (hunk 6 above)
Comment (voddan) [reply cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: Good point, will fix

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt (hunk 4 above)
Comment (voddan) [reply cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: Will fix

File: cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt (hunk 5 above)
Comment (alkopnin) [reply cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: We don't need it as optional for our test scenarios.

Comment (voddan) [reply cordapp/src/test/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/CordaTestBase.kt]: Ok. will change

**Pull request merged**